Core/Core/Logging/SpeckleException.cs

```diff
@@ -1,36 +1,37 @@
 ﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using GraphQL;
+using Sentry.Protocol;
+
 namespace Speckle.Core.Logging
 {
   public class SpeckleException : Exception
   {
+    public List<KeyValuePair<string, object>> GraphQLErrors { get; set; }
     public SpeckleException()
     {
     }
 
-    public SpeckleException(string message) : base(message)
-    {
-    }
-
-    public SpeckleException(string message, Exception inner)
-         : base(message, inner)
-    {
-    }
-  }
-
-
-  public class GraphQLException : Exception
-  {
-    public GraphQLException()
+    public SpeckleException(string message, bool log = false, SentryLevel level = SentryLevel.Info ) : base(message)
     {
+      if (log)
+        Log.CaptureException(this, level);
     }
 
-    public GraphQLException(string message) : base(message)
+    public SpeckleException(string message, GraphQLError[ ] errors, bool log = false,
```

Comment:
Minor comment: maybe log could default to true, since that's the majority of the cases. But it's probably too much hassle to update it, so no biggie!

Reply:
i'd agree re too much hassle :D

Reply:
> cool yes I've incorporated Alan's suggestion! see the notes section of the PR
> 
> can you clarify what you mean? the code snippets in 1. and 2. are the constructors

Yup, soz I replied before looking at the code! I think the only bit missing then is https://github.com/specklesystems/speckle-sharp/pull/278#discussion_r586563243

Reply:
rip soz I assumed there was a reason for dim's suggestion in his comment to default `log = false` but I guess not 😅